### PR TITLE
feat(#82): --verbose, puzzle for server shutdown

### LINKED ISSUE #82: args.rs:28-30: Add --v option for verbose output. Let's...
The puzzle `79-d0a80208` from #79 has to be resolved: 

https://github.com/h1alexbel/fakehub/blob/6932d8bc8d0c334bc7dba4106a820a77e305f622/cli/src/args.rs#L28-L30

The puzzle was created by @rultor on 18-Jul-24. 

Estimate: 25 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by h1alexbel**
@l3r8yJ take a look, please

closes #82

---

## PR-Codex overview
**Focus:** Add verbose output option and integrate tracing library for enhanced logging.

### Detailed summary
- Added `verbose` flag in CLI args for verbose output.
- Integrated `tracing-subscriber` and `tracing` in Cargo.toml for advanced logging.
- Updated server start function to initialize tracing based on `verbose` flag.
- Added test for running in verbose mode.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -33,6 +33,13 @@ async fn main() {
     let args = Args::parse();
     match args.command {
         Command::Start(start_args) => {
+            if start_args.verbose {
```
Comment (l3r8yJ) [inline cli/src/main.rs:36]: @h1alexbel I think that `if verbose { //do logging}` is not the best design solution, i.e. fakehub is a testing tool, and each step should be logged with the `verbose` flag, and this means that we will have to put `if verbose {}` in many places in the a code base that will make it pretty messy. I suggest creating some kind of decorator/proxy/trait that will contain some logic, and this logging entity will decide whether to log or not, wdyt?

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look, please

File: cli/src/main.rs (hunk 1 above)
Comment (h1alexbel) [reply cli/src/main.rs:36]: @l3r8yJ actually it's not. We do `if verbose` only once to subscribe on the log tracings with certain log level, i.e. info or debug. Logging with info or any other level should take us one simple line. WDYT?

Comment (l3r8yJ) [reply cli/src/main.rs:36]: @h1alexbel okay, got it

**Review by l3r8yJ — APPROVED** [review]
@h1alexbel looks good to me now!

File: cli/src/main.rs (hunk 1 above)
Comment (l3r8yJ) [reply cli/src/main.rs:36]: @h1alexbel got it

**Pull request merged**